base/multidimensional.jl

```diff
@@ -487,9 +487,8 @@ rcum_promote_type{T,S<:Number}(op, ::Type{T}, ::Type{S}) = promote_op(op, T, S)
 rcum_promote_type{T<:Number}(op, ::Type{T}) = rcum_promote_type(op, T,T)
 rcum_promote_type{T}(op, ::Type{T}) = T
 
-# handle sums of Vector{Bool} and similar.   it would be nice to handle
-# any AbstractArray here, but it's not clear how that would be possible
-rcum_promote_type{T,N}(op, ::Type{Array{T,N}}) = Array{rcum_promote_type(op,T), N}
+# handle sums of Vector{Bool} and similar.
+rcum_promote_type{T,N}(op, ::Type{<:AbstractArray{T,N}}) = Array{rcum_promote_type(op,T), N}
```

Comment:
Yes, this is a slight increase in functionality (it corrects the rcum type for arrays of non-`Array` arrays of `Bool`).  This is what I originally wanted to do with this function, but at the time it seemed impossible so I punted.  I figured that e.g. calling `cumsum` on e.g. an `Array{BitVector{Bool}}` would be pretty rare.  But it is nice to handle it now that it is easy.

Could be moved to a separate PR, though.